### PR TITLE
updated mock routines handling split between wd and mws_nearby files

### PR DESCRIPTION
**Pull request by forero**
Solves #137

### CODE REVIEW

File: py/desitarget/mock/io.py [hunk 1]
```diff
@@ -321,17 +321,15 @@ def read_wd100pc(mock_dir, target_type, mock_name=None):
     desitarget.io.check_fitsio_version()
     C_LIGHT = 299792.458
 
-    mock_name = 'mock_wd100pc.fits'
+    mock_name = 'mock_100pc.fits'
```
Comment (sbailey) [inline py/desitarget/mock/io.py]: These currently override the `mock_name` input to the function; I think you want
```python
if mock_name is None:
    mock_name = 'mock_100pc.fits'
```

File: py/desitarget/mock/io.py [hunk 2]
```diff
@@ -348,7 +346,70 @@ def read_wd100pc(mock_dir, target_type, mock_name=None):
...
+            DEC positions for the objects in the mock.
+        'Z': :class: `numpy.ndarray`
+            Heliocentric radial velocity divided by the speed of light.
+        'magg': :class: `numpy.ndarray`
+            Apparent magnitudes in Gaia G band
+    """
+    desitarget.io.check_fitsio_version()
+    C_LIGHT = 299792.458
+
+    mock_name = 'mock_wd.fits'
```
Comment (sbailey) [inline py/desitarget/mock/io.py]: Similar to previous comment:
```python
if mock_name is None:
    mock_name = 'mock_wd.fits'
```

File: py/desitarget/mock/selection.py [hunk 3]
```diff
@@ -199,24 +199,37 @@ def mag_select(data, sourcename, targetname, truthname, brick_info=None, density
...
-        # Nearby ('100pc') sample -- everything in the input table that isn't a WD
-        # Expect to refine this in future
-        select_nearby_sample               = (fainter_than_bright_limit) & (brighter_than_faint_limit) & (np.invert(is_wd))
+
+    if(sourcename == 'MWS_NEARBY'):
+        mag_bright = kwargs['mag_bright']
+        mag_faint  = kwargs['mag_faint']
+
+        # Parameters
+        SELECTION_MAG_NAME = 'magg'
```
Comment (sbailey) [inline py/desitarget/mock/selection.py]: Double checking: MWS_WD mocks have "g_sdss" while MWS_NEARBY have "magg"?

Comment (forero) [inline py/desitarget/mock/selection.py]: Right.

File: py/desitarget/mock/io.py (hunk 2 above)
Comment (forero) [inline py/desitarget/mock/io.py]: fixed

File: py/desitarget/mock/io.py (hunk 1 above)
Comment (forero) [inline py/desitarget/mock/io.py]: fixed

File: py/desitarget/mock/io.py [hunk 4]
```diff
@@ -300,6 +300,71 @@ def read_wd100pc(mock_dir, target_type, mock_name=None):
...
+        'magg': :class: `numpy.ndarray`
+            Apparent magnitudes in Gaia G band
+    """
+    desitarget.io.check_fitsio_version()
+    C_LIGHT = 299792.458
+
+    if mock_name is None:
+        filename  = os.path.join(mock_dir,'mock_100pc.fits')
+    else:
+        filename = os.path.join(mock_dir, mock_name+'.fits')
```
Comment (sbailey) [inline py/desitarget/mock/io.py]: details: docstring "name of the mock file" implies that the input should include the extension too, but this adds '.fits'.  I suggest to either not add '.fits' or to check if it is already there before adding.

Comment (forero) [inline py/desitarget/mock/io.py]: I updated the docstring and added a check for '.fits' in the filename

**Comment by moustakas** [comment]
Thank you, @forero, this looks good.

**Pull request merged**